src/crud.js

```diff
@@ -0,0 +1,67 @@
+import Task from './task.js';
+
+// Create tasksList class
+export class TasksList {
+  constructor(name) {
+    this.name = name;
+    this.tasks = [];
+    this.getTasksFromLocalStorage();
+  }
+
+  /*
+  Add local sotrage methods
+    3 main methods added here
+  */
+
+  getTasksFromLocalStorage() {
+    const storedTasks = localStorage.getItem(this.name);
+    if (storedTasks !== undefined && storedTasks !== null) {
+      this.tasks = JSON.parse(storedTasks).map(
+        (task) => new Task(task.description, task.completed, task.id),
+      );
+    }
+    return this.tasks;
+  }
+
+  saveTasksToLocalStorage() {
+    localStorage.setItem(this.name, JSON.stringify(this.tasks));
+  }
+
+  /*
+  Add CRUD methods
+    4 main methods added here
+  */
+  createTask(description) {
+    const error = 'Please provide a description';
+    if (description !== '') {
+      const task = new Task(description, false, this.tasks.length + 1);
+      this.tasks.push(task);
+      this.saveTasksToLocalStorage();
+      return this.tasks;
+    }
+    return error;
+  }
+
+  readTask(id) {
+    return this.tasks.filter((task) => task.id === id);
+  }
+
+  updateDescription(updatedDescrition) {
+    this.description = updatedDescrition;
+  }
+
+  deleteCompletedTasks(id) {
+    const tasks = this.getTasksFromLocalStorage() || [];
+    this.tasksTodelete = [];
+    this.tasksTodelete.push(tasks.filter((task) => task.id === parseInt(id, 10)));
+
+    const filteredTasks = this.tasks.filter((task) => !task.completed);
+    this.saveTasksToLocalStorage();
+    return filteredTasks;
+  }
```

Comment:
- [ ] Function to add new tasks and edit task descriptions has been well implemented. 👍🏻 however there is still a problem with deleting a task, as you can see in the screenshot video below 👇🏻 the targeting book or item is not deleted for example:
- First: I delete the second book2 but all books are deleted from the list
- Second attempt: I delete the second item book 2 but it is book 1 which is deleted from the list instead of the selected  book 2

[Screencast from 05-13-2023 11:24:26 AM.webm](https://github.com/diarisdiakite/To-do-list/assets/79058364/159996e4-5f3d-4df9-8d00-5cc2260ebf9c)

- Kindly fix this 👍🏻